reactor-netty-http/src/main/java/reactor/netty/http/client/HttpClient.java

```diff
@@ -1310,7 +1310,7 @@ public RequestSender request(HttpMethod method) {
 
 	/**
 	 * Specifies the response timeout duration in milliseconds.
-	 * This is time that takes to receive a response after sending a request.
+	 * This is the timeout for the read data operation, i.e., the maximum inactive time.
```

Comment:
Can you clarify `the maximum inactive time`?
The response timeout setting is applied once the request is sent and removed once the response is received. It doesn't apply to the whole request/response.
See https://github.com/reactor/reactor-netty/blob/00201ebb2e696e26b5d2f2df6289a3de86e9cff5/reactor-netty-http/src/main/java/reactor/netty/http/client/HttpClientOperations.java#L557

Reply:
Maybe I misunderstood completely. But here is my interpretation: `ReadTimeoutHandler` that is added in the line you sent is an `IdleStateHandler` and it has a scheduled task for timeouts. The one we use is `readerIdleTimeout` and the task (`ReaderIdleTimeoutTask`) is rescheduled if there is a read operation. So that's why I thought it's read timeout rather than response timeout.
 https://github.com/netty/netty/blob/7b9a97a81f10ea574a8e6f2596075615d2eeb185/handler/src/main/java/io/netty/handler/timeout/IdleStateHandler.java#L482-L513

Reply:
so what's the question?

Reply:
So if my understanding is correct then it should be read timeout instead of response timeout, right? Similar to https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/net/URLConnection.html#setReadTimeout(int).

The response timeout name sounds like the timeout before getting the full response. However, getting the full response may take longer than the given duration in the case of a large payload, for instance.

Reply:
yeah, but we can just extend the javadoc so that it is clear, it is late for this release to change the method name ...

Reply:
Extending the Javadoc sounds good to me. I was just wondering if another method name would be clearer. But I don't have a better suggestion.

Spring's `HttpComponentsClientHttpRequestFactory`, which is for a specific request, also calls it `readTimeout`.
https://docs.spring.io/spring-framework/docs/current/javadoc-api/org/springframework/http/client/HttpComponentsClientHttpRequestFactory.html#setReadTimeout-int-

Reply:
> Spring's HttpComponentsClientHttpRequestFactory, which is for a specific request, also calls it readTimeout.
https://docs.spring.io/spring-framework/docs/current/javadoc-api/org/springframework/http/client/HttpComponentsClientHttpRequestFactory.html#setReadTimeout-int-

isn't this again socket read timeout?

Reply:
Just some rough idea: What do you think about `readOperationInterval ` and then we can describe what `readOperationInterval` means?

```
public final HttpClient responseTimeout(Duration readOperationInterval) {
```

Reply:
Yes, renaming the parameter could work. W.r.t. the suggestion: how about `maxReadOperationInterval`?

Reply:
> Yes, renaming the parameter could work. W.r.t. the suggestion: how about `maxReadOperationInterval`?

sounds OK

Reply:
Proposed something, PTAL.

Reply:
I added some comment, please check them